### ISSUE TITLE
Deployment issues to nodes 

### ISSUE THREAD

**Issue by dribblecastle** [codership/galera-manager-support#16]
Fairly new to this tool.    Since getting this running with 1.4.3 I ran into the following deployment issue. 

The deployment script ran on each of my nodes but stalled on the following 

```
Dec 29, 2021 10:50:23 | host-stdout    | Connecting to github-repository-files.githubusercontent.com (github-repository-files.githubusercontent.com)|185.199.111.154|:443...
Dec 29, 2021 10:50:23 | host-stdout    | connected.
Dec 29, 2021 10:50:23 | host-stdout    | HTTP request sent, awaiting response...
Dec 29, 2021 10:50:23 | host-stdout    | 200 OK
Dec 29, 2021 10:50:23 | host-stdout    | Length: 5730735 (5.5M) [application/x-gzip]
Dec 29, 2021 10:50:23 | host-stdout    | Saving to: ‘latest-linux-amd64.tar.gz’
Dec 29, 2021 10:50:23 | host-stdout    | lates   0%       0  --.-KB/s
Dec 29, 2021 10:50:23 | host-stdout    | latest-linu 100%   5.46M  --.-KB/s    in 0.1s
Dec 29, 2021 10:50:23 | host-stdout    | 2021-12-29 10:50:23 (54.8 MB/s) - ‘latest-linux-amd64.tar.gz’ saved [5730735/5730735]
Dec 29, 2021 10:50:23 | host-stdout    | mysql_wsrep
Dec 29, 2021 10:50:23 | galera-manager | Adding telegraf.conf
Dec 29, 2021 10:50:23 | galera-manager | Adding mysql_wsrep-telegraf-plugin.conf
Dec 29, 2021 10:50:23 | galera-manager | Saving Galera Manager node data
Dec 29, 2021 10:50:23 | galera-manager | Obtaining general_log_file
Dec 29, 2021 10:50:23 | host-stdout    | 172.22.40.90:22$ bash -c 'mysqladmin -u root variables | grep general_log_file | awk '"'"'{print $4}'"'"''
Dec 29, 2021 10:50:23 | galera-manager | failed to execute cluster config script (RunScriptWithConn): command failed (stepName=__step_no_001, commandId=1, commandType=IncludeCommand): command failed (stepName=__step_no_019, commandId=19, commandType=IncludeCommand): command failed (stepName=set_general_log_file, commandId=1, commandType=ExecCommand): failed to set ssh environment variable: ssh: setenv failed
Dec 29, 2021 10:50:23 | galera-manager | Saving Galera Manager node data
```

The workaround was just to delete the node via the UI and then re-add which seems to allow the deployment to complete properly.  (Two checkmarks).    However, when running the nodes I only get CPU info and the logs show this repeatedly.

`Dec 29, 2021 11:28:10 | mariadb-stage01 | /var/log/mysql/error.log | 2021-12-29 11:28:08 1703 [Warning] Access denied for user 'mysqld_exporter'@'127.0.0.1' (using password: YES)`

When viewing the user table I do have the mysqld_exporter user listed.  
`| mysqld_exporter  | localhost       | *XXXXXXXXXXXXXXXXXXX |
`
Seems as if something is not correct with the database access but not sure what to do at this point.  Note, I verified I used the correct DB Root password in my cluster setting.  

FYI, running mariadb10.3 + Ubuntu 20.04.  Locally hosted.

**Comment by denisgcm** [comment]
There is an issue that we should fix. As a temporary workaround I can suggest that you log in to the Galera Manager server via ssh and run `gmc cluster list` (you will have to enter your Galera Manager admin user and password). You will get a similar output:

```
user@ip:~$ gmc cluster list
Enter auth name: admin
Enter auth password:
+----+---------+------+--------------+--------------+-------------+---------------+---------------------+------------------+
| ID |  NAME   | UUID |    SYSTEM    |  DB ENGINE   | HOSTS/NODES | ROOT PASSWORD | MONITORING PASSWORD | BACKUP PASSWORD  |
+----+---------+------+--------------+--------------+-------------+---------------+---------------------+------------------+
|  2 | sample  |      | ubuntu:20.04 | mariadb:10.5 | 3/3         | rootpwd       | monitoringpwd       | backupgpwd       |
+----+---------+------+--------------+--------------+-------------+---------------+---------------------+------------------+
```

Pick your value for `monitoringpwd` and update the password for the `mysqld_exporter` user accordingly. After you do that, the stats should start working.

Please let me know if it helped.

**Comment by dribblecastle** [comment]
Thanks for the quick reply @denisgcm.  I followed your instructions but it didn't work until I created a mySQL user with 127.0.0.1 instead of localhost.   Also, set the permissions according to the previously existing mysqld_exporter:localhost user.   See below.  The good news is things appear to be working as expected.   Looking forward to using this tool as its improves, I just wish I had setup my Galera cluster with it initially so I could use more of the features. 


```
CREATE USER 'mysqld_exporter'@'127.0.0.1'  IDENTIFIED BY 'monitoringpwd';
GRANT SELECT, PROCESS, REPLICATION CLIENT ON *.* TO 'mysqld_exporter'@'127.0.0.1';
```

**Comment by ramtastic** [comment]
We did find some 8081 traffic being blocked and opened that up but I don't think that was our problem (yet).  When the nodes deploy, I can see that MariaDB gets installed on the servers but they do not start from GM.  I can start them individually from SSH but they don't run as a cluster.  It seems that the Galera settings aren't all being configured on deploy.
I saw issue #16 in GitHub and saw that the user had to add a mysqld_exporter user account.  I manually started one of my DB servers and checked the user list and also didn't see that account listed.
